File: backend/src/main/java/com/ecommerce/backend/domain/model/Product.java
package com.ecommerce.backend.domain.model;
import java.math.BigDecimal;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;


@Data
@AllArgsConstructor
@NoArgsConstructor

public class Product {

    private  Integer id;
    private String nombrep;
    private String codigo;
    private String descripcion;
    private String urlImagen;
    private BigDecimal precio;
    private LocalDateTime fechaCreacion;
    private LocalDateTime fechaActualizacion;
    private Integer userId;
    private Integer categoriaId;

}

File: frontend/frontend/src/app/common/product.ts
export class Product {
    constructor(
        public id:number,
        public nombrep:string,
        public codigo:string,
        public descripcion:string,
        public precio:number,
        public urlImagen:string,
        public image:File,
        public userId:string,
        public categoriaId:string,
        
    ){}
}
